### ISSUE TITLE
Users on standard image

### ISSUE THREAD

**Issue by ajrouvoet** [tu-os-course/TI2726C-Assignments-2016#4]
Source editing, compilation and the such should all be doable as a non-root user

In addition, steps to become root + when and why to be root should be added to the manual.

**Comment by jackodirks** [comment]
I don't think it should. Using the GPIO pins is hardware interfacing and linux only allows hardware interfacing to be done by the root user, or a root user process. In my opinion this should not be violated to allow the wiringPi calls to be done by non-root users.

Because of the nature of the course where one of the important goals is to learn about what the OS does and why it does it, I advice against allowing non-root users to use hardware without a syscall.

On a more technical note: the wiringPi libs call modprobe to load some kernel modules and to do that without root would mean to fork the linux kernel. The alternative would be to let users login as root, which is also a viliolation to what Linux stands and dies for, but easier doable. I also advice against that because this is a very good practical example of what we are getting told during the lectures: only one user can change hardware: the root.

Just my two cents.

**Comment by ajrouvoet** [comment]
I agree, "everything" is too much and not what I intended. Edited.
